### PR TITLE
test: add tests for descriptive error message in ipa user-add

### PR DESCRIPTION
**Pull request by b3lix**
Add tests for renaming existing user and group with invalid name or only numeric name
Related: https://pagure.io/freeipa/issue/9378

### CODE REVIEW

**Comment by b3lix** [comment]
Hi @miskopo
yes I am already working on that second point, resolving conflicts

File: ipatests/test_xmlrpc/test_group_plugin.py [hunk 1]
```diff
@@ -182,7 +183,7 @@ def test_create_with_name_starting_with_numeric(self):
         testgroup.create()
         testgroup.delete()
 
-    def test_create_with_numeric_only_group_name(self):
+    def test_create_with_numeric_only_groupname(self):
         """Try to create a group with name only contains numeric chars"""
         testgroup = GroupTracker(
             name=u'1234', description=u'Numeric only group name',
```
Comment (rcritten) [inline ipatests/test_xmlrpc/test_group_plugin.py:189]: should be invalidgroup2 now that it has its own variable.

File: ipatests/test_xmlrpc/test_user_plugin.py [hunk 2]
```diff
@@ -53,6 +53,7 @@
 
 invaliduser1 = u'+tuser1'
 invaliduser2 = u''.join(['a' for n in range(256)])
+invaliduser3 = u'1234'
```
Comment (rcritten) [inline ipatests/test_xmlrpc/test_user_plugin.py:56]: test_create_with_numeric_only_username should use invaliduser3 instead of hardcoding u'1234'

**Comment by rcritten** [comment]
The commit message refers to adding tests for renames only. The user changes include adding a numeric-only stage user and other smaller mods.

File: ipatests/test_xmlrpc/test_group_plugin.py [hunk 3]
```diff
@@ -193,6 +194,27 @@ def test_create_with_numeric_only_group_name(self):
         )):
             testgroup.create()
 
+    def test_rename_to_invalid_groupname(self, group):
+        """ Try to rename group using an invalid name """
+        group.ensure_exists()
+        command = group.make_update_command(
+            updates=dict(setattr=u'cn=%s' % invalidgroup1))
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/test_group_plugin.py:201]: The proper method to rename a group is to use `ipa group-mod name --rename newname`, which would be translated into
```
        command = group.make_update_command(
            updates=dict(rename=invalidgroup1))
        with raises_exact(errors.ValidationError(
            name='rename',
            error=ERRMSG_GROUPUSER_NAME.format('group'),
        )):
            command()

```

File: ipatests/test_xmlrpc/test_group_plugin.py [hunk 4]
```diff
@@ -193,6 +194,27 @@ def test_create_with_numeric_only_group_name(self):
...
+            name='group_name',
+            error=ERRMSG_GROUPUSER_NAME.format('group'),
+        )):
+            command()
+
+    def test_rename_to_numeric_only_groupname(self, group):
+        """ Try to rename group using an invalid numeric only name """
+        group.ensure_exists()
+        command = group.make_update_command(
+            updates=dict(setattr=u'cn=%s' % invalidgroup2))
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/test_group_plugin.py:212]: Same comment as above:
```
        command = group.make_update_command(
            updates=dict(rename=invalidgroup2))
        with raises_exact(errors.ValidationError(
            name='rename',
            error=ERRMSG_GROUPUSER_NAME.format('group'),
        )):
            command()
```

**Review by flo-renaud — COMMENTED** [review]
Hi @b3lix 
Thanks for the PR. Please find inline comments.

File: ipatests/test_xmlrpc/test_group_plugin.py (hunk 3 above)
Comment (b3lix) [reply ipatests/test_xmlrpc/test_group_plugin.py:201]: Thank you for this suggestion, before I was inspired by test_rename, but this better.

Comment (flo-renaud) [reply ipatests/test_xmlrpc/test_group_plugin.py]: @rcritten agreed, but unfortunately using setattr doesn't go through the validation code and lets you use an invalid value. We should open a ticket to fix that.

Comment (flo-renaud) [reply ipatests/test_xmlrpc/test_group_plugin.py]: Opened https://pagure.io/freeipa/issue/9396 to track the issue and PR https://github.com/freeipa/freeipa/pull/6883 with a fix + test for the ipa group-mod --setattr cn= part

Comment (b3lix) [reply ipatests/test_xmlrpc/test_group_plugin.py]: Thank you @flo-renaud for dealing with it.

**Review by flo-renaud — COMMENTED** [review]
Hi @b3lix 
please find inline comments.

File: ipatests/test_xmlrpc/test_group_plugin.py [hunk 5]
```diff
@@ -183,40 +183,39 @@ def test_create_with_name_starting_with_numeric(self):
...
         )
         with raises_exact(errors.ValidationError(
             name='group_name',
             error=ERRMSG_GROUPUSER_NAME.format('group'),
         )):
             testgroup.create()
 
-    def test_rename_setattr_to_invalid_groupname(self, group):
-        """ Try to rename group using an invalid name with settatr cn= """
+    def test_rename_to_invalid_groupname(self, group):
```
[the remark this answers is not in the document]
Comment (b3lix) [reply ipatests/test_xmlrpc/test_group_plugin.py:197]: Hi @flo-renaud this action was done by me after [discussion above](https://github.com/freeipa/freeipa/pull/6874#pullrequestreview-1476338130) , but I will re-add those 2 using settatr and also keep those using rename method. So we will have both methods covered, please feel free to correct me if I am wrong.

File: ipatests/test_xmlrpc/test_user_plugin.py [hunk 6]
```diff
@@ -504,22 +504,24 @@ def test_rename_to_invalid_login(self, user):
...
-                name='rename',
-                error=ERRMSG_GROUPUSER_NAME.format('user'))):
+            name='rename',
+            error=ERRMSG_GROUPUSER_NAME.format('user'),
+        )):
             command()
 
-    def test_rename_setattr_to_numeric_only_username(self, user):
-        """ Try to change name to name with only numeric chars with setattr"""
+    def test_rename_to_numeric_only_username(self, user):
```
[the remark this answers is not in the document]
Comment (b3lix) [reply ipatests/test_xmlrpc/test_user_plugin.py:512]: Same as [above](https://github.com/freeipa/freeipa/pull/6874#discussion_r1251799679).

**Comment by flo-renaud** [comment]
Thanks for the updated patch, works for me.

**Comment by miskopo** [comment]
master:

* 4a3e3efb84cee9e3784246f3bc47f1f56b266bc0 test: add tests for descriptive error message in ipa user-add

**Pull request closed without merging**